### ISSUE TITLE
Flush table for unit tests running

### ISSUE THREAD

**Issue by santaux** [yandex/ClickHouse#1252]
Hi guys,

Could tell me please, if there is a way to flush/clean tables completely (not to delete one or several record and just to remove everything completely?)? I try to write some unit tests for application which uses ClickHouse. But as you know normally after each test case (or group of test cases) each table should be cleared to do not interrupt next test cases running.

I was thinking about dropping all tables and building them each time from the SQL dump, but I think that's not the best way to do it. May be you could give me some advises how to resolve my problem?

Thanks.

**Comment by ludv1x** [comment]
You could use `ALTER TABLE [db.]table DROP PARTITION 'month'` command to clear all data from a specified month (for example '201709') (according to partitioning key).

**Comment by santaux** [comment]
Ok, thank you! I'll try it!

**Comment by victor-perov** [comment]
Hello, for now, I found in addition these possible ways:

- `ALTER TABLE [db.]table DELETE WHERE filter_expr`, so you can do `ALTER TABLE tbl DELETE WHERE 1 = 1`
- `TRUNCATE TABLE tbl`

Though, @alexey-milovidov I found a situation, when insert right after `TRUNCATE TABLE ...` provides sometimes strange results, it's like partially inserted (or deleted) data or something like this.

Let me describe how it happens.

For functional tests I would like to populate my tables with data. Everything lives in dockerized containers, which is fine. For executing test I could continue with two possible approaches:

1. drop ClickHouse container, and recreate it again (with applying all of my migrations). As soon as it up and healthy, I could do population my tables with data and, finally, execute tests.
2. do not drop clickhouse container, but drop data in tables, populate with new data and execute tests.

Second way gives me huge difference in time, and I'd like to continue with it, but during truncating data, I faced with some behaviour I don't expect.

I do tests against tables with MergeTree engine. Is it expected, that truncate goes fast, but skip acknowledgement, and we have to wait, before populating data?

Probably I missing something, or `TRUNCATE TABLE` should be used for different scenario.

ClickHouse version 18.14.17 revision 54409

**Comment by filimonov** [comment]
@victor-perov it's quite hard to follow/react comments apperared in closed issue :) 

Could you please open new issue about problematic TRUNCATE behavior? 
Does the problem reproduces?
How big the inserts are?
Can you share some sample of problematic sequence? I.e. create-tuncate-insert-data partially loss?
